rotkehlchen/globaldb/upgrades/v3_v4.py

```diff
@@ -256,17 +258,69 @@ def _add_optimism_contracts(cursor: 'DBCursor', eth_scan_abi_id: int, multicall_
     log.debug('Exit _add_optimism_contracts')
 
 
+def _copy_assets_from_packaged_db(
+        cursor: 'DBCursor',
+        assets_ids: list[str],
+        root_dir: Path,
+) -> None:
+    """
+    Copy missing assets from the packaged globaldb to the globaldb
+    Since the upgrade happens when the connection to the globaldb hasn't been created yet we need
+    to manually copy the assets information as we don't have the GlobalDBHandler object.
+
+    We need to commit the changes here because otherwise the packaged globaldb is busy and we
+    can't detach it. At this point all the information in the upgrade should be fine.
+
+    TODO: This approach assumes that the schema in the packaged globaldb and the schema in v3 are
+    the same. When this doesn't hold anymore we should keep a copy of the old schema or ask users
+    to use an intermediate version to upgrade.
+    """
+    packaged_db_path = root_dir / 'data' / 'global.db'
+    identifiers_quotes = ",".join("?" * len(assets_ids))
+    cursor.execute(f'ATTACH DATABASE "{packaged_db_path}" AS packaged_db;')
+    cursor.execute(f'INSERT INTO assets SELECT * FROM packaged_db.assets WHERE identifier IN ({identifiers_quotes});', assets_ids)  # noqa: E501
+    cursor.execute(f'INSERT INTO evm_tokens SELECT * FROM packaged_db.evm_tokens WHERE identifier IN ({identifiers_quotes});', assets_ids)  # noqa: E501
+    cursor.execute(f'INSERT INTO common_asset_details SELECT * FROM packaged_db.common_asset_details WHERE identifier IN ({identifiers_quotes});', assets_ids)  # noqa: E501
+    cursor.execute('COMMIT;')
+    cursor.execute('DETACH DATABASE packaged_db;')
+
+
 def _add_assets_collections(cursor: 'DBCursor') -> None:
     """
     Insert in the global db the assets collection properties for the assets we know and
     create the collections.
+
+    If any of the assets that needs to go in the collections is missing we copy it from the
+    packaged globaldb.
     """
     log.debug('Enter _add_assets_collections')
     root_dir = Path(__file__).resolve().parent.parent.parent
+    # regex used to identify assets that will be added to the collections
+    asset_regex = re.compile(r'eip155[a-zA-F0-9:\/]+')
     with open(root_dir / 'data' / 'assets_collections.sql') as f:
         raw_sql_sentences = f.read()
         per_table_sentences = raw_sql_sentences.split('\n\n')
         for sql_sentences in per_table_sentences:
+            # check if we are adding the assets
+            if 'multiasset_mappings' in sql_sentences:
```

Comment:
Why not just split the sql multiassets mapping into its own sql file and just do this after the for loop? This way you separate concerns better and don't have to do if checks for each loop iteration

Reply:
went with the other approach improving the check

Reply:
What other approach? I don't understand. You still have an if check here